TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift

```diff
@@ -76,32 +76,66 @@ class TradeItPortfolioAccountDetailsViewController: TradeItViewController, Trade
             )
         }
 
-        let positionsPromise = Promise<Void> { fulfill, reject in
+        let positionsAndQuotesPromise = self.positionsPromise(linkedBrokerAccount: linkedBrokerAccount).then { portfolioPositions in
+            return self.quotesPromise(portfolioPositions: portfolioPositions)
+        }
+
+        authenticatePromise.then { _ in
+            return when(fulfilled: accountOverviewPromise, positionsAndQuotesPromise)
+        }.then { _, positions in
+            self.tableViewManager.updatePositions(withPositions: positions)
+        }.always {
+            onRefreshComplete()
+        }.catch { error in
+            print(error)
+        }
+    }
+
+    // MARK: Private
+
+    private func positionsPromise(linkedBrokerAccount: TradeItLinkedBrokerAccount) -> Promise<[TradeItPortfolioPosition]> {
+        return Promise<[TradeItPortfolioPosition]> { fulfill, reject in
             linkedBrokerAccount.getPositions(
                 onSuccess: { positions in
-                    self.tableViewManager.updatePositions(withPositions: positions)
-                    fulfill()
+                    fulfill(positions)
                 },
                 onFailure: { error in
-                    self.tableViewManager.updateAccount(withAccount: nil)
+                    self.tableViewManager.updatePositions(withPositions: nil)
                     reject(error)
                 }
             )
         }
+    }
 
-        firstly {
-            authenticatePromise
-        }.then { _ in
-            return when(fulfilled: accountOverviewPromise, positionsPromise)
-        }.catch { error in
-            print(error)
-        }.always {
-            onRefreshComplete()
+    private func quotesPromise(portfolioPositions: [TradeItPortfolioPosition]) -> Promise<[TradeItPortfolioPosition]> {
+        let symbols = portfolioPositions
+            .filter { $0.position?.lastPrice == nil }
+            .map { $0.position?.symbol }
+            .flatMap { $0 }
```

Comment:
Is it possible when position is nil to end with a nil object in the `symbols` array ?

Reply:
The `flatMap` should remove it but I'll double check

Reply:
I can probably combine the `map` and `flatMap` actually